hypervisor/arch/x86/guest/instr_emul.c

```diff
@@ -214,12 +214,10 @@ static const struct instr_emul_vie_op one_byte_opcodes[256] = {
 
 #define	GB				(1024 * 1024 * 1024)
 
-static uint64_t size2mask[9] = {
-	[1] = 0xffUL,
-	[2] = 0xffffUL,
-	[4] = 0xffffffffUL,
-	[8] = 0xffffffffffffffffUL,
-};
+static inline uint64_t size2mask(uint8_t bytes)
+{
+	return ((1UL << (bytes << 3U)) - 1UL);
+}
```

Comment:
When bytes = 8, this operation may overflow. So, let us keep use array, but change the oxfffffff to what I mentioned in email,